crates/curp/src/server/raw_curp/mod.rs

```diff
@@ -601,6 +601,20 @@ impl<C: Command, RC: RoleChange> RawCurp<C, RC> {
         self.ctx.id_barrier.wait_all(conflict_cmds)
     }
 
+    /// Wait all logs in previous term have been applied to state machine
+    pub(super) fn wait_no_op_applied(&self) -> Box<dyn Future<Output = ()> + Send + Unpin> {
+        // if the leader is at term 1, it won't commit a no-op log
+        if self.term() == 1 {
```

Comment:
how to understand that *if the leader is at term 1, it won't commit a no-op log* ?

Reply:
In Xline we specify the initial leader at startup, the no-op log is only append on vote success. On the other hand, if the leader is at term 1, there's no other log in previous term, there's no need to wait for a no-op log to be committed.